optuna/integration/sklearn.py

```diff
@@ -513,11 +513,12 @@ def _estimator_type(self) -> str:
 
     @property
     def best_index_(self) -> int:
-        """Index which corresponds to the best candidate parameter setting."""
+        """Trial number which corresponds to the best candidate parameter setting.
 
-        df = self.trials_dataframe()
+        Retuned value is equivant to ``optuna_search.best_trial_.number``.
```

Comment:
```suggestion
        Retuned value is equivant to ``OptunaSearchCV.best_trial_.number``.
```

It's not a strong opinion but I prefer `OptunaSearchCV.best_trial_.number` over `optuna_search.best_trial_.number` since `optuna_search` is not defined in this scope. Do you have any intention to use `optuna_search` here?

Reply:
I followed the corresponding part of [scikit-learn's `GridSearchCV`](https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.GridSearchCV.html):

> The dict at `search.cv_results_['params'][search.best_index_]` gives the parameter setting for the best model, that gives the highest mean score (`search.best_score_`).

Reply:
I see. Thank you for the clarification!